src/Shared/PackageManagers/NuGetProjectManager.cs

```diff
@@ -337,267 +214,210 @@ public override Uri GetPackageAbsoluteUri(PackageURL purl)
         /// <inheritdoc />
         public override async Task<PackageMetadata> GetPackageMetadata(PackageURL purl, bool useCache = true)
         {
-            PackageMetadata metadata = new();
-            string? content = await GetMetadata(purl, useCache);
-            if (string.IsNullOrEmpty(content)) { return metadata; }
+            string latestVersion = (await EnumerateVersions(purl, useCache)).First();
 
-            // convert NuGet package data to normalized form
-            JsonDocument contentJSON = JsonDocument.Parse(content);
-            JsonElement root = contentJSON.RootElement;
-            JsonElement latestCatalogEntry = GetLatestCatalogEntry(root);
+            // Construct a new PackageURL that's guaranteed to have a version, the latest version is used if no version was provided.
+            PackageURL purlWithVersion = !string.IsNullOrWhiteSpace(purl.Version) ? 
+                purl : new PackageURL(purl.Type, purl.Namespace, purl.Name, latestVersion, purl.Qualifiers, purl.Subpath);
 
-            metadata.Name = latestCatalogEntry.GetProperty("id").GetString();
-            metadata.Description = latestCatalogEntry.GetProperty("description").GetString();
-            
-            // Title is different than name or description for NuGet packages, not always used.
-            // metadata.Title = GetLatestCatalogEntry(root).GetProperty("title").GetString();
+            NuGetMetadata? packageVersion =
+                await _nuGetProvider.GetMetadataAsync(purlWithVersion, useCache: useCache);
+
+            if (packageVersion is null)
+            {
+                throw new NullReferenceException();
+            }
+
+            PackageMetadata metadata = new();
+
+            metadata.Name = packageVersion.PackageId;
+            metadata.Description = packageVersion.Description;
 
-            metadata.PackageManagerUri = ENV_NUGET_ENDPOINT;
+            metadata.PackageManagerUri = ENV_NUGET_ENDPOINT_API;
             metadata.Platform = "NUGET";
             metadata.Language = "C#";
-            // metadata.SpokenLanguage = GetLatestCatalogEntry(root).GetProperty("language").GetString();
-            metadata.PackageUri = $"{metadata.PackageManagerUri}/packages/{metadata.Name?.ToLower()}";
-            metadata.ApiPackageUri = $"{RegistrationEndpoint}{metadata.Name?.ToLower()}/index.json";
+            metadata.PackageUri = $"{ENV_NUGET_HOMEPAGE}/{metadata.Name?.ToLowerInvariant()}";
+            metadata.ApiPackageUri = $"{RegistrationEndpoint}{metadata.Name?.ToLowerInvariant()}/index.json";
```

Comment:
Can we actually tolerate `null` for `metadata.Name` here? Or would these property values just be garbage if that was the case?

If these would be garbage, would it make sense to use a non-nullable value (maybe `packageVersion.PackageId`), or to assert that the value is not null?

Reply:
Resolved in the other PR.